metron-deployment/scripts/platform-info.sh

```diff
@@ -75,28 +79,32 @@ npm --version
 
 # C++ compiler
 echo "--"
-echo "g++"
-g++ --version
-
-# C++11 compliant compiler
-echo "--"
-OBJFILE=/tmp/test
-CPPFILE=/tmp/test.cpp
-cat > $CPPFILE <<- EOM
+if [[ $(command -v g++) && $(g++ --version 2>/dev/null) ]]; then
+  g++ --version
+  # C++11 compliant compiler
+  echo "--"
+  OBJFILE=/tmp/test
+  CPPFILE=/tmp/test.cpp
+  cat > $CPPFILE <<- EOM
 #include <iostream>
 using namespace std;
 int main() {
     cout << "Hello World!" << endl;
     return 0;
 }
 EOM
-g++ -std=c++11 $CPPFILE -o $OBJFILE
-if [ $? -eq 0 ]; then
-    echo "Compiler is C++11 compliant"
+  g++ -std=c++11 $CPPFILE -o $OBJFILE &>/dev/null
+  if [ $? -eq 0 ]; then
+      echo "Compiler is C++11 compliant"
+  else
+      echo "Warning: Compiler is NOT C++11 compliant"
+  fi
+  rm -f $CPPFILE $OBJFILE
+elif [[ $(command -v g++) ]]; then
+  echo "Warning: g++ not properly configured"
```

Comment:
I'm not following what exactly "g++ not properly configured" means based on the conditions here.  Under what conditions would I be able to run `command -v g++`, but not `g++ --version 2>/dev/null`?

Reply:
I ran into a situation where `g++` was in my path on a fresh macOS box, but I needed to accept a license via `xcode-select --install` before it could be used in any way, including calling `--version`.

Reply:
No, don't worry about it.  

Right now, I am just on the fence as to whether we need the extra complexity with this PR.  Seems like the simpler approach does output any problems that the user might run into.

Reply:
@ottobackwards Do you prefer this over the (subjectively) simpler approach that I took in the original PR?  I prefer the simpler approach, but I know I am often wrong. :)

Reply:
Don't make me comment on the bash skills of the two people I look up to the most!

Reply:
That being said, @JonZeolla's approach it more inline with what I would expect intuitively, as seen by my very early comments on the pr.  I wouldn't go to the mattresses to get it done that way, but seeing has he went and did the work, I would do it that way.

Reply:
I'm fine with that.  @JonZeolla Can you open your PR on apache/metron?  You can use the same JIRA.  I'll close my PR.

Reply:
You did all the real work, how about we just merge mine into yours and not attribute to me going into Metron

Reply:
I can't merge this PR.  There's a lot of other cruft included.  But I'll add what you did to the original and push it out.

Reply:
I added this to the original branch.  Closing the PR.  Thanks @JonZeolla

Reply:
Hmm.    All of the other stuff in my PR was just a merge of master at the time.  But ok cool thanks